docs/source/tutorials/fashion_mnist.ipynb

```diff
@@ -0,0 +1,1258 @@
+    "            loss.backward()\n",
+    "            optimizer.step()\n",
+    "\n",
+    "            # print statistics\n",
+    "            running_loss += loss.item()\n",
+    "\n",
+    "        accuracy = get_test_accuracy(model, testloader)\n",
+    "        print(\n",
+    "            f\"epoch: {epoch + 1} loss: {running_loss / len(trainloader):.3f} test acc: {accuracy}\"\n",
+    "        )\n",
+    "\n",
+    "        running_loss = 0.0\n",
+    "\n",
+    "        if accuracy > best_test_accuracy:\n",
+    "            best_epoch = epoch\n",
+    "\n",
+    "        if epoch - best_epoch > patience:\n",
+    "            print(f\"Early stopping at epoch {epoch + 1}\")\n",
+    "            break\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "# Method for computing out-of-sample embeddings\n",
+    "def compute_embeddings(model, testloader):\n",
+    "    embeddings_list = []\n",
+    "\n",
+    "    with torch.no_grad():\n",
+    "        for data in tqdm(testloader):\n",
+    "            images, labels = data[\"image\"].to(device), data[\"label\"].to(device)\n",
+    "\n",
+    "            embeddings = model.embeddings(images)\n",
+    "            embeddings_list.append(embeddings.cpu())\n",
+    "\n",
+    "    return torch.vstack(embeddings_list)\n",
+    "\n",
+    "\n",
+    "# Method for computing out-of-sample predicted probabilities\n",
+    "def compute_pred_probs(model, testloader):\n",
+    "    pred_probs_list = []\n",
+    "\n",
+    "    with torch.no_grad():\n",
+    "        for data in tqdm(testloader):\n",
+    "            images, labels = data[\"image\"].to(device), data[\"label\"].to(device)\n",
+    "\n",
+    "            outputs = model(images)\n",
+    "            pred_probs_list.append(outputs.cpu())\n",
+    "\n",
+    "    return torch.vstack(pred_probs_list)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Create K folds of the dataset for cross validation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "K = 3  # Set to small value here to ensure quick runtimes, we recommend 5 or 10 in practice for more accurate estimates.\n",
+    "\n",
+    "kfold = KFold(n_splits=K, shuffle=True, random_state=0)\n",
+    "splits = kfold.split(transformed_dataset)\n",
+    "\n",
+    "train_id_list, test_id_list = [], []\n",
+    "\n",
+    "for fold, (train_ids, test_ids) in enumerate(splits):\n",
+    "    train_id_list.append(train_ids)\n",
+    "    test_id_list.append(test_ids)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Compute out-of-sample predicted probabilities and feature embeddings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_epochs = 3\n",
+    "patience = 2\n",
+    "batch_size = 4\n",
```

Comment:
You NEED to have comment for every parameter you have set to some artificially low toy setting to ensure quick tutorial runtime. If users do not change this value in practice, they will get bad results! 

Do not expect users to know anything about ML, the tutorial needs to explain everything. For references/inspiration, see the comment in other tutorials about such parameters (and the one about K K-fold xval above)

Reply:
also group all such parameters together in one place, so user can just change them all in one place (in this case group these with the `K` parameter above).

Reply:
Added helpful comments and also grouped them together